kt-android-runtime/runtime/src/main/java/io/sdkgen/runtime/SdkgenHttpClient.kt

```diff
@@ -0,0 +1,229 @@
+                        InternalResposne(
+                            JsonObject().apply {
+                                addProperty("type", "Fatal")
+                                addProperty("message", "Erro Fatal (502) - Tente novamente")
+                            },
+                            null
+                        )
+                    )
+                    return@suspendCoroutine
+                }
+
+                val responseBody = try {
+                    val stringBody = response.body?.string()
+                    gson.fromJson(stringBody, JsonObject::class.java)
+                } catch (e: Exception) {
+                    e.printStackTrace()
+                    continuation.resume(
+                        InternalResposne(
+                            JsonObject().apply {
+                                addProperty("type", "Fatal")
+                                addProperty("message", "Erro de serialização")
+                            },
+                            null
+                        )
+                    )
+                    return@suspendCoroutine
+                }
+
+                val pref = applicationContext.getSharedPreferences("api", Context.MODE_PRIVATE)
+                pref.edit().putString("deviceId", responseBody.get("deviceId").asString).apply()
+
+                if (!responseBody.get("ok").asBoolean) {
+                    val jsonError = responseBody.getAsJsonObject("error")
+                    continuation.resume(InternalResposne(jsonError, null))
+                } else {
+                    continuation.resume(InternalResposne(null, responseBody))
+                }
+                response.close()
+            } catch (e: Exception) {
+                httpTimer.cancel()
+                e.printStackTrace()
+                continuation.resume(
+                    InternalResposne(
+                        JsonObject().apply {
+                            if (e is SocketTimeoutException || e is InterruptedException) {
+                                addProperty("type", "Connection")
+                                addProperty("message", "Conexão excedeu o tempo de espera.")
+                            } else {
+                                addProperty("type", "Fatal")
+                                addProperty("message", "Chamada falhou sem mensagem de erro!")
+                            }
+                        },
+                        null
+                    )
+                )
+            }
+        } catch (e: Exception) {
+            e.printStackTrace()
+            continuation.resume(
+                InternalResposne(
+                    JsonObject().apply {
+                        addProperty("type", "Fatal")
+                        addProperty("message", "Ocorreu um erro desconhecido na conexão.")
+                    },
+                    null
+                )
+            )
+        }
```

Comment:
This seems to be the only client that creates custom errors with custom strings based on response, unsure if we should do that. @lbguilherme?

Reply:
I think it is ok, but it would be better to create a new error type just for that. Fatal is more for unknown server conditions. An internet connectivity issue should be handled by the app

Reply:
I'll add this as an enhancement issue later